optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,53 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.2.0")
+class BestValueStagnationEvaluator(BaseImprovementEvaluator):
+    """
+    Evaluates the stagnation period of the best value in an optimization process.
+
+    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
+    and is designed toevaluate the remaining trials before reaching this maximum period
+    of allowed stagnation.
+
+    Args:
+        - max_stagnation_trials (int): The maximum number of trials allowed for stagnation.
+
+    Example Usage:
+        >>> evaluator = BestValueStagnationEvaluator(max_stagnation_trials=30)
+        >>> stagnated_trials = evaluator.evaluate()
```

Comment:
Could you follow the Optuna's  docstring style, please?

https://github.com/optuna/optuna/blob/9f0be552b618c695afd5569c81fb45140c1613f2/optuna/terminator/improvement/evaluator.py#L52-L62

Reply:
Updated as follow.
```
    """Evaluates the stagnation period of the best value in an optimization process.

    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
    and is designed toevaluate the remaining trials before reaching this maximum period
    of allowed stagnation.

    Args:
        max_stagnation_trials (int):
            The maximum number of trials allowed for stagnation.
    """
```